metadata/src/main/java/org/apache/kafka/metadata/authorizer/StandardAuthorizerData.java

```diff
@@ -391,11 +392,13 @@ static AuthorizationResult findResult(Action action,
                     if (!IMPLIES_DESCRIBE_CONFIGS.contains(acl.operation())) return null;
                     break;
                 default:
-                    if (!action.operation().equals(acl.operation())) return null;
+                    if (acl.operation() != ALL && action.operation() != acl.operation()) {
```

Comment:
Hmm, would it be easier to just put
```
if (acl.operation() != ALL) {
}
```

around the whole block?

So
```
if (acl.operation() != ALL) {
  if (acl.permissionType().equals(ALLOW)) {
       .... switch stuff ...
    } else {
        if (!action.operation().equals(acl.operation())) return null;
    }
}
```

Reply:
Heh. Yeah, makes sense.